### ISSUE TITLE
Support token authentication from clouds.yaml

### ISSUE THREAD

**Issue by leesdolphin** [gophercloud/utils#25]
I'm using Terraform to deploy my instances and I want to store my authentication token in my `clouds.yaml` so I don't have to keep authenticating with my password. 

The `clouds.yaml` looks like this and works with the openstack CLI:

```yaml
clouds:
  example:
    auth_type: token
    auth:
      auth_url: https://auth.cloud.example.com:5000/v3
      token: my-secret-token
      project_id: 1234
      project_name: "example.net"
    region_name: "region"
    interface: "public"
    identity_api_version: 3
```

I've talked with some of our cloud's team about making documentation improvements to openstack itself.

**Comment by jtopjian** [comment]
@leesdolphin Thank you for reporting this. It looks like token auth is indeed possible with `clouds.yaml`:

https://github.com/openstack/os-client-config/blob/480bfc27224a675a483a3c00a2c831054a1ffdc6/os_client_config/tests/test_config.py#L755-L768

I'll look at getting this added.

**Comment by jtopjian** [comment]
@leesdolphin This has been implemented in #27. It'll be supported in Terraform shortly. I apologize for the delay.
